test/common/stats/stat_test_utility.cc

```diff
@@ -1,10 +1,26 @@
 #include "test/common/stats/stat_test_utility.h"
 
+#include "common/memory/stats.h"
+
 namespace Envoy {
 namespace Stats {
 namespace TestUtil {
 
-void foreachStat(int num_clusters, std::function<void(absl::string_view)> fn) {
+bool hasDeterministicMallocStats() {
+  // We can only test absolute memory usage if the malloc library is a known
+  // quantity. This decision is centralized here. As the preferred malloc
+  // library for Envoy is TCMALLOC that's what we test for here. If we switch
+  // to a different malloc library than we'd have to re-evaluate all the
+  // thresholds in the tests referencing hasDeterministicMallocStats().
+#ifdef TCMALLOC
+  const std::string str("surely this will have to allocate some memory");
+  return Memory::Stats::totalCurrentlyAllocated() != 0;
```

Comment:
LGTM, though I suppose since this is a `const std::string` initialized from a string literal it *could* conceivably not allocate anything on the heap. It might be worth trying internally too, to make sure you get the expected results.